### PR TITLE
Suggest to combine C# and VB sections into a single .NET Guide

### PR DESCRIPTION
**Pull request by lucasnobredev**
## Summary
This PR came out from this [issue](https://github.com/dotnet/docs/issues/6181). 

Well, I started a single page like @rpetrusha asked. There was two moments that I had to mention "Blabla for C# or Bleble for Visual Basic". I'm not sure it was great, but I believe it was simple and comprehensive.

The links on final page are not correct **YET**. 

About "ms.assetid", I'm not sure what I have to write.

## Suggested reviewers
@rpetrusha

### CODE REVIEW

File: docs/standard/assemblies-and-the-global-assembly-cache.md
Comment (rpetrusha) [inline docs/standard/assemblies-and-the-global-assembly-cache.md]: Along with the See also section, add a section here:
\## In this section
\[.NET assembly file format](assembly-format.md)

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thank you so much, @lucasnobredev, for beginning the work of consolidating information on .NET assemblies. In addition to my specific comments, there are a number of general comments:
- Create a subdirectory of **standard** named **assembly** and put the file there. We'll probably want to have other assembly-related files there as the consolidation effort makes progress.
- The file assembly-format.md should also be moved from the standard directory to standard/assembly.
- Because the assembly-format.md file has been moved, we should redirect from the old URL to the new one. Redirects are stored in the root's .openpublishing.redirection.json file. I think the format is clear, but don't hesitate to ask. 
- The TOC needs to be modified to include the new content. It's found in docs/toc.md. Currently, the line that reads:
```
## [.NET Assembly File Format](standard/assembly-format.md)
```
should be replaced with
```
## [Assemblies in .NET](standard/assembly/index.md)
### [.NET Assembly File Format](standard/assembly/assembly-format.md)
```
- The two original files in the VB and C# subdirectories should eventually be deleted. Since they're in index.md files, though, that poses some difficulty, since child items will no longer appear. What we thought was the best approach is this:
   - We delete the assembly-gac/index.md files for VB and C# and add them to the redirection file.
   - We add sections to this topic that read something like "Other information on assemblies for C# developers" and "Other information on assemblies for Visual Basic developers".
   - We include a link to each of the topics in the C# and VB assembly-gac directory.

  We can then delete them gradually, add redirects, and revise this topic as they're removed.

I've still not compared the C# and VB versions of the topic to see if the VB version has some unique content. I'll do that on the next review of your PR.

Again, @lucasnobredev, thank you so much for undertaking this project.

**Comment by rpetrusha** [comment]
I failed to mention, @lucasnobredev, a more general issue -- that in the case of assemblies, we want to try to revise topics that reflect development with .NET Core as well as .NET Framework. Much of this content was written for .NET Framework 1.0 or 1.1 and so is also badly dated.

**Comment by lucasnobredev** [comment]
I'll do, @rpetrusha . Sorry for taking so long.

**Comment by mairaw** [comment]
@lucasnobredev just following up on this one. If you can't make the changes, we'll do it for you. Just let us know.

**Comment by lucasnobredev** [comment]
@rpetrusha @mairaw hello guys, I'm sorry for taking so long. I made some changes asked in review, but I also have some doubts:

- About "Because the assembly-format.md file has been moved [...]", what old URL are you talking about? C# and VB pages? There is not anything in .openpublishing.redirection.json.

- About "We can then delete them gradually, add redirects, and revise this topic as they're removed.", are you talking about C# and VB pages?

- The section "In this section" will be divide in two which will specifing C# and VB links?

**Comment by lucasnobredev** [comment]
Ohh, also I received this for email. I think @mairaw already commented about not using "~" for links.. Should I change?

image

**Comment by mairaw** [comment]
Sorry @lucasnobredev, missed your question during the holidays. You can use the ~ links but we prefer the relative links. The problem here might be that the files don't exist anymore or the links are wrong. Let me take a look.

**Review by mairaw — CHANGES_REQUESTED** [review]
@lucasnobredev left some comments that should fix the links but also some extra review comments that should probably be addressed before merging this. Thanks!

**Comment by lucasnobredev** [comment]
@mairaw I fixed the last one. About the first one, the media folder it's in _standard/media_. Should I bring this to _standard/assembly_?  About the second one, I have no idea about what it is. Can you help me?

image

Ohh, there something else: it's still not so clear for me about "See also" section. I wrote just about C# articles. Should I join C# and VB files?

File: docs/toc.md
Comment (mairaw) [inline docs/toc.md]: one more suggestion: let's rename this article to either file-format or just format, since our SEO tools usually complain when we have repeated words in the path.

**Comment by mairaw** [comment]
Yes, the new article should be a joint view of both articles. Hopefully, as we move content into a single target, we'd reduce the list of links. Also, as part of this or on a separate PR, it'd be good to review the incoming links to the articles we're removing to update links and link text.

**Comment by lucasnobredev** [comment]
@mairaw I'll look for wrong links.

About C#/VB articles from "See Also" section, I believe it would be better if we use all existing articles in this section, merge it this PR and after that, we can look to all C#/VB articles which has the same content to convert to unique files. Unless... we could hold this PR, open new ones to fix all content and merge it all in one time.

What do you think?

**Comment by mairaw** [comment]
Either way is fine @lucasnobredev. We can just do this one article now and start working on the next ones in separate PRs knowing that the see also section from the "new" article would change as we bring the other topics from the same section.
Even though having them all move together is ideal, it might become a giant PR if we do so.

**Comment by lucasnobredev** [comment]
@mairaw I couldn't be able to fix a problem on file-format file. The build service said there is a problem in:

Line : [Warning] 1 invalid cross reference(s) "<xref:System.Reflection.Assembly.ReflectionOnlyLoadFrom?displayProperty=nameWithType%2A>".

Could you help me?

After that, I think we're able to merge it. I already wrote at issue the others files to unify. Once we get done here, I'll open new PRs.

**Comment by rpetrusha** [comment]
@lucasnobredev, the correct format of the xref is  \<xref:System.Reflection.Assembly.ReflectionOnlyLoadFrom%2A?displayProperty=nameWithType>

**Comment by lucasnobredev** [comment]
@rpetrusha Done!

**Comment by BillWagner** [comment]
@rpetrusha @mairaw (Going through old PRs) It looks like this is ready for a final review..

File: docs/standard/assembly/index.md
Comment (rpetrusha) [inline docs/standard/assembly/index.md]:
```suggestion
-   A file table describing all the other files that make up the assembly, such as other assemblies you created that your .exe or .dll file relies on, or even bitmap or Readme files.  
```

File: docs/standard/assembly/index.md
Comment (rpetrusha) [inline docs/standard/assembly/index.md]:
```suggestion
-   An *assembly reference list*, which is a list of all external dependencies — .dlls or other files your application needs that may have been created by someone else. Assembly references contain references to both global and private objects. Global objects are available to all other applications. In .NET Core, they are coupled with a particular .NET Core runtime. In .NET Framework, they reside in the global assembly cache. The <xref:System.IO?displayProperty=nameWithType> namespace is an example of an assembly in the global assembly cache. Private objects must be in a directory at either the same level as or below the directory in which your application is installed.  
```

File: docs/standard/assembly/index.md
Comment (rpetrusha) [inline docs/standard/assembly/index.md]:
```suggestion
 Because assemblies contain information about content, versioning, and dependencies, the applications that use them need not rely don Windows registry values to function properly. Assemblies reduce .dll conflicts and make your applications more reliable and easier to deploy. In many cases, you can install a .NET-based application simply by copying its files to the target computer. For more information, see [Assembly Manifest](../../framework/app-domains/assembly-manifest.md).  
```

File: docs/standard/assembly/index.md
Comment (rpetrusha) [inline docs/standard/assembly/index.md]:
```suggestion

> [!NOTE]
> Most assemblies from the .NET Class Library are referenced automatically. In some cases, though, a system assembly may not automatically be referenced. In .NET Core, you can add a reference to the NuGet package that contains the assembly either by using NuGet Package Manager in Visual Studio or by adding a [<PackageReference>](../../core/tools/dependencies.md#the-new-packagereference-element) element for the assembly to the *.csproj or *.vbproj project. In .NET Framework, you can add a reference to the assembly by using the **Add Reference** dialog in Visual Studio or by using the `-reference` command line option for the [C#](../../csharp/language-reference/compiler-options/reference-compiler-option.md) or [Visual Basic](../../visual-basic/reference/command-line-compiler/reference.md) compilers.

```

File: docs/standard/assembly/index.md
Comment (rpetrusha) [inline docs/standard/assembly/index.md]:
```suggestion
 Compile your application by building it in Visual Studio, by building it from the command line by using .NET Core command-line interface (CLI) tools, or by building .NET Framework assemblies with a command-line compiler. For more information about building assemblies using .NET CLI tools, see [.NET Core command-line interface (CLI) tools](../../core/tools/index.md). For building assemblies with the command-line compilers, see [Command-line build with csc.exe](../../csharp/language-reference/compiler-options/command-line-building-with-csc-exe.md)  for C# and [Building from the Command Line](../../visual-basic/reference/command-line-compiler/building-from-the-command-line.md) for Visual Basic.  
```

**Review by rpetrusha — COMMENTED** [review]
Thanks for continuing to work on this PR, @lucasnobredev. I've suggested a number of changes and additions. Once they're made, your PR will be ready to merge.

**Comment by lucasnobredev** [comment]
@rpetrusha @BillWagner @mairaw Done!

**Review by rpetrusha — APPROVED** [review]

**Review by mairaw — APPROVED** [review]
Thank you for all your work on this @lucasnobredev!

**Pull request merged**
